### PR TITLE
cartodb.createLayer selects the first data layer instead of assuming that it's in position 1

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes: #593 

We're assuming that the layer in position #1 was a `layergroup` or `namedmap`. This PR implements a smarter way of looking for the first data layer.

cc/ @xavijam

### CODE REVIEW

File: src/api/layers.js [hunk 1]
```diff
@@ -96,12 +96,21 @@
...
-          return;
+        var index = options.layerIndex;
+        if (index !== undefined) {
+          if(visData.layers.length <= index) {
+            promise.trigger('error', 'layerIndex out of bounds');
+            return;
+          }
+          layerData = visData.layers[index];
+        } else {
+          var DATA_LAYER_TYPES = ['namedmap', 'layergroup'];
```
Comment (xavijam) [inline src/api/layers.js]: ['namedmap', 'layergroup'] Didn't we have those variables somewhere?

**Comment by xavijam** [comment]
Sounds good!

File: src/api/layers.js (hunk 1 above)
Comment (alonsogarciapablo) [inline src/api/layers.js]: Probably! I'm going to find a better place for these guys. Thanks!

**Comment by javisantana** [comment]
<3

**Pull request merged**